fission/src/systems/simulation/wpilib_brain/WPILibBrain.ts

```diff
@@ -286,18 +289,24 @@ export class PWMGroup extends SimOutputGroup {
     }
 
     public Update(_deltaT: number) {
-        // let average = 0;
+        let average = 0;
         for (const port of this.ports) {
             const speed = SimPWM.GetSpeed(`${port}`) ?? 0
-            // average += speed;
+            average += speed;
             console.log(port, speed)
         }
-        // average /= this.ports.length
-
-        // this.drivers.forEach(d => {
-        //     (d as WheelDriver).targetWheelSpeed = average * 40
-        //     d.Update(_deltaT)
-        // })
+        average /= this.ports.length
+
+        this.drivers.forEach(d => {
+            if (d instanceof WheelDriver) {
+                d.targetWheelSpeed = average * 40
+            } else if (d instanceof HingeDriver) {
+                d.targetVelocity = average * 40
+            } else if (d instanceof SliderDriver) {
+                d.targetVelocity = average * 40
```

Comment:
Im assuming the reason you didn't just do 
```ts
} else if (d instanceof HingeDriver || d instanceof SliderDriver) {
```
is because you are going to change it in the future to be unique

Reply:
Might be better to do a switch statement, since this is looping and switch statements don't use comparisons which are quite slow when looped because of branch predicting. This probably won't matter, but might since this is in an update function that will only be called more often as more bodies are spawned.

Reply:
I didn't realize they had the same name until after I'd separated the clauses but I'll change that to not repeat itself.

Reply:
> Might be better to do a switch statement, since this is looping and switch statements don't use comparisons which are quite slow when looped because of branch predicting. This probably won't matter, but might since this is in an update function that will only be called more often as more bodies are spawned.

The only way I know of to make this a switch statement is
```ts
switch (true) {
  case d instanceof WheelDriver:
    // wheel
  case d instanceof HingeDriver:
  case d instanceof SliderDriver:
    // hinge and slider
}
```

I'm not sure if this has the same benefits of a regular switch statement, since the cases would still need to be evaluated at runtime so I don't think it could make a jump table. I also think this syntax is less immediately understandable.

We would also need to make it possible to simulate multiple robots at once for this to be impacted by the number of robots spawned, which I don't think is currently supported by our websocket implementation.

Reply:
Oh you're right, a jump table probably isn't possible for whatever operation instanceof does since it doesn't match to a value 👍

Reply:
The only way to do something like this would be to create a mapping object, then jumping with d.constructor.name, which seems excessive, I'll keep this in mind incase this update function ever becomes a bottleneck though.

Reply:
[Seems like your implementation would be more efficient though](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html)

We could also match directly on `d.constructor` if d will never be a subclass of these drivers
https://stackoverflow.com/questions/36332665/how-to-use-instanceof-in-a-switch-statement